src/map.cpp

```diff
@@ -0,0 +1,18 @@
+
+#include "../include/map.hpp"
+#include "../include/tower.hpp"
+#include "../include/enemy.hpp"
+#include "../include/waypoint.hpp"
+
+Map::Map(Tower *towers, Enemy *enemies, Waypoint *start)
```

Comment:
I'm assuming these will have to resize in the future, so why don't we use vector?

http://en.cppreference.com/w/cpp/container/vector